toolbox/ci2gubernator/lib/ci2gubernator.go

```diff
@@ -129,21 +122,18 @@ func (c *Converter) UploadBuildLog(logFile string) error {
 
 // UpdateLastBuildTXT updates latest-build.txt to GCS
 func (c *Converter) UpdateLastBuildTXT() error {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
-	defer cancel()
-	m, err := gcslock.New(ctx, c.bucket, lastBuildTXT)
+	gcsPath := filepath.Join(filepath.Dir(c.gcsPathPrefix), lastBuildTXT)
+	update := func() error {
+		log.Printf("Updating gs://%s to be %d", gcsPath, c.build)
+		return c.gcsClient.Write(gcsPath, fmt.Sprintf("%d", c.build))
+	}
+	exists, err := c.gcsClient.Exists(gcsPath)
 	if err != nil {
 		return err
 	}
-	if err = m.ContextLock(ctx); err != nil {
```

Comment:
seems easier to read with a more linear flow:


if exists && needUpdate() {
  log.Printf("Updating gs://%s to be %d", gcsPath, c.build)
  return c.gcsClient.Write(gcsPath, fmt.Sprintf("%d", c.build))
}

and then implement needUpdate() as a helper method...

Reply:
Done